src/shell/components/InAppAnnouncement/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState, useMemo } from "react";
+import { ThemeProvider } from "@mui/material";
+import { theme } from "@zesty-io/material";
+import moment from "moment";
+import { useLocalStorage } from "react-use";
+import {
+  Dialog,
+  DialogContent,
+  DialogActions,
+  Typography,
+  Stack,
+  Button,
+  Box,
+} from "@mui/material";
+import OpenInNewRoundedIcon from "@mui/icons-material/OpenInNewRounded";
+import ScheduledRoundedIcon from "@mui/icons-material/ScheduleRounded";
+import PlayArrowRoundedIcon from "@mui/icons-material/PlayArrowRounded";
+
+import { useGetAnnouncementsQuery } from "../../services/marketing";
+
+export const InAppAnnouncement = () => {
+  const [activeAnnouncementIndex, setActiveAnnouncementIndex] = useState(0);
+  const [readAnnouncements, setReadAnnouncements] = useLocalStorage(
+    "zesty:readAnnouncements",
+    []
+  );
+  const { data: announcements } = useGetAnnouncementsQuery();
+
+  const unreadAnnouncements = useMemo(() => {
+    if (announcements?.length) {
+      return announcements
+        .filter((announcement) => {
+          const isInPublishRange = moment().isBetween(
+            moment(announcement?.start_date_and_time),
+            moment(announcement?.end_date_and_time)
+          );
+
+          return (
+            isInPublishRange && !readAnnouncements.includes(announcement?.zuid)
+          );
+        })
+        ?.sort((a, b) => moment(b.created_at).diff(a.created_at));
+    }
+  }, [announcements]);
+
+  const onIgnoreAnnouncement = (zuid: string) => {
+    if (!readAnnouncements.includes(zuid)) {
+      setReadAnnouncements([...readAnnouncements, zuid]);
+      setActiveAnnouncementIndex(activeAnnouncementIndex + 1);
+    }
+  };
+
+  const announcementData = unreadAnnouncements
+    ? unreadAnnouncements[activeAnnouncementIndex]
+    : null;
+
+  return (
+    <ThemeProvider theme={theme}>
+      {announcementData && (
+        <Dialog
+          open
+          onClose={() => onIgnoreAnnouncement(announcementData?.zuid)}
+          maxWidth="md"
+          PaperProps={{ sx: { width: 640 } }}
+          data-cy="AnnouncementPopup"
+        >
+          <DialogContent sx={{ p: 0 }}>
+            <Stack
+              m={2.5}
+              p={3}
+              width={600}
+              height={340}
+              sx={{
+                background: "linear-gradient(90deg, #EC4A0A 0%, #FD853A 100%)",
+              }}
+              alignItems="center"
+              justifyContent="center"
+              boxSizing="border-box"
+            >
+              <Box
+                component="img"
+                alt="announcement-banner-image"
+                src={announcementData?.feature_image?.data[0]?.url ?? ""}
```

Comment:
This URL should contain `width` and `height` parameters  that match the rendered size.